### PR TITLE
libSQL serverless driver for TypeScript and JavaScript

### LINKED ISSUE #4: Serverless JavaScript driver
The client API we have is great for cases where you can do `LD_PRELOAD`, but other databases are adopting serverless JavaScript drivers such as:

https://github.com/planetscale/database-js

https://developers.cloudflare.com/d1/platform/client-api/

Let's implement a JavaScript driver that connects to `sqld`.

### PR DESCRIPTION
**Pull request by penberg**
Fixes #4

### CODE REVIEW

File: packages/libsql-serverless/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+const db = new Database();
+await db.execute("CREATE TABLE users (email TEXT)");
+await db.execute("INSERT INTO users (email) VALUES ('alice@example.com')");
+await db.execute("INSERT INTO users (email) VALUES ('bob@example.com')");
+```
+
+## Features
+
+* SQLite JavaScript API
+* SQLite-backed local-only backend
```
Comment (haaawk) [inline packages/libsql-serverless/README.md:17]: SQLite or LibSQL?

Comment (penberg) [reply packages/libsql-serverless/README.md:17]: SQLite right now. We want to switch to libSQL, of course, but there's no npm package for that.

File: packages/libsql-serverless/src/lib/libsql-js.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+} from "sqlite3";
+
+/**
+ * A SQL query result set.
+ */
+type ResultSet = {
+  /**
+   * Query results.
+   */
+  results: unknown[] | null;
```
[the remark this answers is not in the document]
Comment (penberg) [reply packages/libsql-serverless/src/lib/libsql-js.ts:16]: It's an array of rows, but the rows are not typed. If you have better typing in mind, send a patch to fix it up or propose something :)

**Comment by penberg** [comment]
bors merge

**Pull request merged**